lib/alchemy/controller_actions.rb

```diff
@@ -46,16 +46,8 @@ def alchemy_user_signed_in?
     # Returns the current site.
     #
     def current_alchemy_site
-      @current_alchemy_site ||= begin
-        site_id = params[:site_id] || session[:site_id]
-        if site_id.nil?
-          session.delete :site_id
-          Site.find_for_host(request.host)
-        else
-          session[:site_id] = site_id
-          Site.find(site_id)
-        end
-      end
+      raise(ActionController::RoutingError, 'Not Found') if params[:site_id] || session[:site_id]
```

Comment:
I thought this too, but then I followed your suggestion :stuck_out_tongue_closed_eyes: